architecture/adr-list/adr-003-cli-tools.md

```diff
@@ -0,0 +1,125 @@
+
+Ideally, the `cheqd-node` project would provide a consistent set of CLI tools rather than two separate tools with varying feature sets between them.
+
+This ADR will focus on the CLI tool architecture choice for `cheqd-node`.
+
+## Context
+
+### Assumptions / Considerations
+
+#### **Likelihood of introducing bugs or security vulnerabilities**
+
+1. Any CLI tool architecture chosen should not increase the likelihood of introducing bugs, security vulnerabilities, or design pattern deviations from upstream Cosmos SDK.
+2. Actions that are carried out on ledger through a CLI tool in `cheqd-node` now include token functionality as well as identity functionality. E.g., if a DID gets compromised, there could be mechanisms to recover or signal that fact to issuers/verifiers. If tokens or the staking balance of node operators get compromised, this may potentially have more severe consequences for them.
+
+#### User / Node Operator preferences
+
+1. Would node operators want a single CLI to manage everything?
+   1. This might be the case with node operators from an SSI / digital identity background, or node operators familiar with Hyperledger Indy CLI / VDR Tools CLI.
+   2. A “single CLI” could be a single tool as far as the user sees, but actually consist of multiple modules beneath it in how it’s implemented.
+2. Would node operators be okay with having two separate CLIs?
+   1. One for Cosmos-ledger functions, and one for identity-specific functions.
+   2. Unlike existing Hyperledger Indy networks, it is anticipated that some of the node operators on the cheqd network will have experience running Cosmos validator nodes. For this group, having to learn a new “single” CLI tool could cause a steeper learning curve and a worse user experience than what they have now.
+   3. Node operators may want one/separate CLIs for security and operational reasons, i.e., for a separation of concerns in terms of functionality.
+
+### Existing CLI architecture overview
+
+Currently, there are two separate CLI tools being used by the development team to interact with the ledger: Cosmos-based CLI and VDR CLI.
+
+![Current Cosmos CLI and VDR tools CLI architecture](https://lh3.googleusercontent.com/cMdfEe19vqDVaRJ0kP97KGCUHauEpnh2TV1OhmvGqOFqqIkhWXGkdKxONDLjW2rnU83k9yelFWK_jhsqQoF57tNf8ChrPeIZsiLys3LKVT_QKG9Gk7Mir4ChbCeiUKs2V7l7jE8d=s0)
+
+### Existing CLI tools feature matrix
+
+| Only available in Cosmos CLI | Decision needed whether these are in Cosmos CLI or VDR CLI | Only available in VDR CLI |
+| :--- | :--- | :--- |
+| Cosmos transactions + Queries | Signing service + Key storage | Identity transactions + Queries |
+| MultiSig | \(Transaction + Query\) sending + Proof validation | DIDs + VCs \(+ DID storage\) |
+| Network bootstrapping commands |  |  |
```

Comment:
This state seems to be before the decision. Can we have a table showing the feature split after the decision? The second column then won't be "decision needed", it would be become something like "Available in both..."

Reply:
Moved to the decisions section too cause it's more about decision